### PR TITLE
Kubernetes Alternator Demo

### PR DESCRIPTION
**Pull request by dahankzter**
Play Tic Tac Toe!

### CODE REVIEW

File: kubernetes-alternator-demo/operator.yaml [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        image: scylladb/scylla-operator:v0.1.8
```
Comment (mmatczuk) [inline kubernetes-alternator-demo/operator.yaml:344]: We could use the new operator.

File: kubernetes-alternator-demo/cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  cpuset: true
+  sysctls:
+    - "fs.aio-max-nr=2097152"
+  datacenter:
+    name: <gcp_zone>
+    racks:
+      - name: <gcp_region>
+        scyllaConfig: "scylla-config"
+        scyllaAgentConfig: "scylla-agent-config"
+        members: 3
```
Comment (mmatczuk) [inline kubernetes-alternator-demo/cluster.yaml:97]: We have 4 scylla k8s nodes  and we use only 3 out of 4.

File: kubernetes-alternator-demo/cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        resources:
+          limits:
+            cpu: 30
+            memory: 115G
+        placement:
+          nodeAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+              nodeSelectorTerms:
+                - matchExpressions:
+                  - key: failure-domain.beta.kubernetes.io/zone
```
Comment (mmatczuk) [inline kubernetes-alternator-demo/cluster.yaml:110]: This is nice we should really have a helm chart for that.

File: kubernetes-alternator-demo/operator.yaml (hunk 1 above)
Comment (dahankzter) [reply kubernetes-alternator-demo/operator.yaml:344]: Sure we can bump it.

File: kubernetes-alternator-demo/cluster.yaml (hunk 2 above)
Comment (dahankzter) [reply kubernetes-alternator-demo/cluster.yaml:97]: It just takes so long to boot them....

File: kubernetes-alternator-demo/operator.yaml (hunk 1 above)
Comment (dahankzter) [reply kubernetes-alternator-demo/operator.yaml]: Fixed.

**Comment by mmatczuk** [comment]
Please add a newline and maybe a prefix to distinguish "Press any key to continue" from normal stdout

```
scylla-prom-prometheus-alertmanager-5fc99b4c9f-b2ft4   0/2     ContainerCreating   0          15s
scylla-prom-prometheus-node-exporter-29cd7             1/1     Running             0          16s
scylla-prom-prometheus-pushgateway-59d7f789b-cdl72     0/1     Running             0          15s
scylla-prom-prometheus-server-68b77d4fff-v7p99         0/2     ContainerCreating   0          15s
Now we need to deploy the Scylla Operator. Press any key to continue
```

**Comment by mmatczuk** [comment]
I'm lacking a script to delete gke resources created with gke.sh.

**Comment by mmatczuk** [comment]
"Let's look a little at the Operator deploymentkubectl get pods -n scylla-operator-system" - lack of space

**Comment by mmatczuk** [comment]
Ingress IP is assigned but not detected by script

```
 michal@lh:scylla-operator-eval/eks (master)$ kubectl describe -n scylla service tictactoe 
Name:                     tictactoe
Namespace:                scylla
Labels:                   <none>
Annotations:              service.beta.kubernetes.io/aws-load-balancer-backend-protocol: (https|http|ssl|tcp)
Selector:                 game=tictactoe
Type:                     LoadBalancer
IP:                       10.55.242.153
LoadBalancer Ingress:     34.82.64.112
Port:                     <unset>  80/TCP
TargetPort:               5000/TCP
NodePort:                 <unset>  31709/TCP
Endpoints:                10.52.6.21:5000
Session Affinity:         None
External Traffic Policy:  Cluster
Events:
  Type    Reason                Age    From                Message
  ----    ------                ----   ----                -------
  Normal  EnsuringLoadBalancer  9m22s  service-controller  Ensuring load balancer
  Normal  EnsuredLoadBalancer   8m39s  service-controller  Ensured load balancer
```

It says 

```
spec:
  clusterIP: 10.55.242.153
  externalTrafficPolicy: Cluster
  ports:
  - nodePort: 31709
    port: 80
    protocol: TCP
    targetPort: 5000
  selector:
    game: tictactoe
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - ip: 34.82.64.112
```

**Comment by mmatczuk** [comment]
./grafana-web-ui.sh is only called in bootstrap.sh that is not used.

bootstrap.sh should be removed and ./grafana-web-ui.sh should be migrated to webinar.sh.

**Comment by mmatczuk** [comment]
grafana-web-ui.sh should direct to `localhost` not `0.0.0.0`

**Comment by dahankzter** [comment]
> I'm lacking a script to delete gke resources created with gke.sh.

I think this one should go into the operator repo.

**Comment by dahankzter** [comment]
> `grafana-web-ui.sh` is only called in `bootstrap.sh` that is not used.
> 
> bootstrap.sh should be removed and ./grafana-web-ui.sh should be migrated to webinar.sh.

These are simple convenience scripts. The other webinar also contain them. What I used to do was run the bootstrap in good time of the webinar since it takes quite some time to get it up then during the webinar I launched the webinar.sh script as shim on top of the existing scylla cluster. We can join them but I had it that way first and it was too inconvenient.

**Comment by dahankzter** [comment]
> Ingress IP is assigned but not detected by script
> 
> ```
>  michal@lh:scylla-operator-eval/eks (master)$ kubectl describe -n scylla service tictactoe 
> Name:                     tictactoe
> Namespace:                scylla
> Labels:                   <none>
> Annotations:              service.beta.kubernetes.io/aws-load-balancer-backend-protocol: (https|http|ssl|tcp)
> Selector:                 game=tictactoe
> Type:                     LoadBalancer
> IP:                       10.55.242.153
> LoadBalancer Ingress:     34.82.64.112
> Port:                     <unset>  80/TCP
> TargetPort:               5000/TCP
> NodePort:                 <unset>  31709/TCP
> Endpoints:                10.52.6.21:5000
> Session Affinity:         None
> External Traffic Policy:  Cluster
> Events:
>   Type    Reason                Age    From                Message
>   ----    ------                ----   ----                -------
>   Normal  EnsuringLoadBalancer  9m22s  service-controller  Ensuring load balancer
>   Normal  EnsuredLoadBalancer   8m39s  service-controller  Ensured load balancer
> ```
> 
> It says
> 
> ```
> spec:
>   clusterIP: 10.55.242.153
>   externalTrafficPolicy: Cluster
>   ports:
>   - nodePort: 31709
>     port: 80
>     protocol: TCP
>     targetPort: 5000
>   selector:
>     game: tictactoe
>   sessionAffinity: None
>   type: LoadBalancer
> status:
>   loadBalancer:
>     ingress:
>     - ip: 34.82.64.112
> ```

I don't understand. Doesn't the script eventually find the IP? For me it could take quite some time to get it.

**Comment by mmatczuk** [comment]
Ad grafana-web-ui.sh

It should be added to main webinar flow IMO

Ad Ingress

The script runs forever while it's ready

**Pull request merged**